CMake/config.h.cmake.in

```diff
@@ -0,0 +1,218 @@
+/* Took config/config.h.in and cmakeified */
+
+/* Define to dummy `main' function (if any) required to link to the Fortran
+   libraries. */
+#cmakedefine FC_DUMMY_MAIN
+
+/* Define if F77 and FC dummy `main' functions are identical. */
+#cmakedefine FC_DUMMY_MAIN_EQ_F77
+
+/* Define to a macro mangling the given C identifier (in lower and upper
```

Comment:
I cleaned this up. The new methodology creates on pascal:
```
#define FC_FUNC(name,NAME)  name ## _
#define FC_FUNC_(name,NAME)  name ## _
```

FC_FUNC is defined in root CMakeLists.txt based on FortranCInterface_GLOBAL_XXXX vars generated by CMake.

Reply:
Ok, I think that means that the fortran compiler on pascal mangles names using lower case and appends an underscore so `DBPutMaterial(...)` becomes `dbputmaterial_(...)` for fortran.

In looking at Silo for fortran support, in `src/silo/silo_f.h` we appear to be doing two brands of name mangling, *manually* with our own definitions of `FC_FUNC` there. For example, in Fortran, `DBPutMaterial(...)` (C function name) becomes shortened to `DBPUTMAT(...)` or variants.

It would be best if we just handled this the *normal* (CMake) way. But, I think if we did, we'd break all existing Fortran callers because that second brand of name mangling (e.g. shortening the names...I have no idea why we did that except maybe back in the Fortran 77 days, we wanted only 6 or 8 character names), is what they are all coded to use.

My inclination would be to somehow *deprecate* the shortened names at least by no longer documenting them (we'd update the documentation) but still always keep them in place for stragglers who don't update.

Thoughts?

Reply:
I'm really not familiar with Fortran and C name mangling. When I was looking into cleaning up the cmake code, I looked at what was created in config.h by calling configure.  When I got the same FC_FUNC definitions with CMake, I thought I was on the right track.  Anything you want to do regarding this is fine with me.

Reply:
If interested, [here](https://en.wikipedia.org/wiki/Name_mangling#Fortran) is some background on Fortran name mangling. 

Ok, I think I'd like to have @brugger1 weigh in on this too. 

@brugger1 I don't think I've looked at this too closely before but Silo's fortran interface seems to do mangling at two levels. One is some kind of manual *shortening* of the C function names (e.g. `DBPutMaterial()` becomes `DBPUTMAT()`). The other is the standard Fortran mangling (case and underscore). This means we explicitly maintain the *shortening* mangling in [`src/silo/silo_f.h`](https://github.com/LLNL/Silo/blob/main/src/silo/silo_f.h).

I don't know why we did this in the past but the question is...can we get rid of it? Or, at the very least, *deprecate* it. I don't think Fortran users are necessarily expecting the *shortening* mangling and it would be more natural for them to have to deal with only the *normal* Fortran mangling.

I am thinking best strategy is...

- Do not introduce any new *shortened* mangled Fortran symbol names. All new Fortran symbol names will be handled *normally* (e.g. just normal Fortran mangling via CMake `FC_FUNC`).
- Deprecate all *shortened* Fortran mangled names (e.g. issues a warning message once per execution when used)
- Silently maintain *shortened* mangled names but do not document them
- `silo.inc` and friends has both just *normal* mangled names and the *shortened* mangled names.
- Remove *shortened* mangled names at some point in the future.

 is to silently support *normal* Fortran mangling and then document it and eliminate documentation for the legacy (shortened) mangling but nonetheless